walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -89,22 +105,126 @@ else if (!event.getWhen().contains(time)){
     return getTime(events, requestDuration, meetingTimes);
   }
 
-  private Collection<TimeRange> getOptionalTime(Collection<Event> events, long requestDuration, Collection<TimeRange> requiredTimes) {
+  private List<TimeRange> getOptionalTime(Collection<Event> events, long requestDuration, List<TimeRange> requiredTimes) {
     return getTime(events, requestDuration, requiredTimes);
   }
 
-  private int eventType(Event event, MeetingRequest request) {
-    HashSet<String> attendees = new HashSet<> (event.getAttendees());
-    attendees.retainAll(request.getAttendees());  
-    if (attendees.size() > 0) {
-      return 1;
-    }
-    HashSet<String> optionalAttendees = new HashSet<> (event.getAttendees());
-    optionalAttendees.retainAll(request.getOptionalAttendees());
-    if (optionalAttendees.size() > 0) {
-        return 2;
+  private boolean hasRequiredAttendees(Event event, MeetingRequest request) {
+    Collection<String> eventAttendees = event.getAttendees();
+    Collection<String> meetingAttendees = request.getAttendees();  
+    boolean hasAttendees = !Collections.disjoint(eventAttendees, meetingAttendees);
+    return hasAttendees;
+  }
+
+  private boolean hasOptionalAttendees(Event event, MeetingRequest request) {
+    Collection<String> eventAttendees = event.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    boolean hasAttendees = !Collections.disjoint(eventAttendees, optionalAttendees);
+    return hasAttendees;
+  }
+
+  private List<TimeRange> getTimeAlternative(Collection<Event> events, long requestDuration, List<TimeRange> initial) {
```

Comment:
Please make a separate PR with the alternative implementation. I think this PR is close to done, and you should submit it first.